cpp/velox/substrait/SubstraitToVeloxPlan.cc

```diff
@@ -1875,8 +1877,10 @@ void SubstraitToVeloxPlanConverter::createNotEqualFilter(
   // Value < upper
   std::unique_ptr<FilterType> upperFilter;
   if constexpr (std::is_same_v<RangeType, common::BigintRange>) {
-    upperFilter = std::make_unique<common::BigintRange>(
-        getLowest<NativeType>() /*lower*/, notVariant.value<NativeType>() - 1 /*upper*/, nullAllowed);
+    if (getLowest<NativeType>() <= notVariant.value<NativeType>() - 1) {
```

Comment:
If the filter was like `a > -32768`
it used to add `[-32768, -32769]` range which can produce wrong result as it will match values [-32678] so added a check for validity of range. `lower <= upper`

Reply:
@mskapilks  is UT for this already present, if not we should add this?

Reply:
it is present GlutenUnwrapCastInComparisonEndToEndSuite